glean-core/src/metrics/string_list.rs

```diff
@@ -66,25 +66,29 @@ impl StringListMetric {
             value
         };
 
+        let mut error = None;
         glean
             .storage()
             .record_with(&self.meta, |old_value| match old_value {
                 Some(Metric::StringList(mut old_value)) => {
                     if old_value.len() == MAX_LIST_LENGTH {
-                        //let msg = format!(
-                        //    "String list length of {} exceeds maximum of {}",
-                        //    old_value.len() + 1,
-                        //    MAX_LIST_LENGTH
-                        //);
-                        // TODO (bug 1557828) - record the error when it doesn't deadlock.
-                        // record_error(glean, &self.meta, ErrorType::InvalidValue, msg);
+                        let msg = format!(
+                            "String list length of {} exceeds maximum of {}",
+                            old_value.len() + 1,
+                            MAX_LIST_LENGTH
+                        );
+                        error = Some(msg);
                         return Metric::StringList(old_value);
```

Comment:
I find the early return inside an anonymous function a bit confusing.

Maybe move the `old_value.push(value.clone())` into an `else` clause, but always return `Metric::StringList(old_value)` in the same place at the end of the expression?